scylla/src/transport/cluster.rs

```diff
@@ -235,23 +248,34 @@ impl ClusterData {
         let mut all_nodes: Vec<Arc<Node>> = Vec::with_capacity(metadata.peers.len());
 
         for peer in metadata.peers {
+            let translated_peer_address = match address_translator {
+                Some(ref translator) => match translator.translate_address(&peer).await {
+                    Ok(addr) => addr,
+                    Err(err) => {
+                        warn!("Could not translate address; TranslationError: {:?}", err);
```

Comment:
If the translator is present, we shouldn't assume that untranslated addresses are safe to connect (apart from initial contact nodes). Here, IMO, we should skip the node and inform about it in the warning message.